src/main/java/com/minecolonies/coremod/commands/AbstractCommandParser.java

```diff
@@ -0,0 +1,678 @@
+        {
+            final List<String> tabCompletions = possibleActionCommands.keySet().stream().filter(k -> k.startsWith(lowerCaseArg0)).collect(Collectors.toList());
+            if (possibleActionCommands.containsKey(lowerCaseArg0))
+            {
+                final ActionMenuHolder actionMenuHolder = possibleActionCommands.get(lowerCaseArg0);
+                final ActionArgument actionArgument = actionMenuHolder.getActionArgument();
+                parsedActionArgumentList.add(actionArgument);
+                return new ParsingResult(tabCompletions, actionMenuTreeNode, parsedActionArgumentList, actionMenuState, (String) null);
+            }
+            else
+            {
+                return new ParsingResult(tabCompletions, actionMenuTreeNode, parsedActionArgumentList, actionMenuState, lowerCaseArg0);
+            }
+        }
+        final ActionMenuHolder holder = possibleActionCommands.get(lowerCaseArg0);
+        if (null == holder)
+        {
+            final List<String> tabCompletions = Collections.emptyList();
+            return new ParsingResult(tabCompletions, actionMenuTreeNode, parsedActionArgumentList, actionMenuState, lowerCaseArg0);
+        }
+        else
+        {
+            possibleActionCommands.remove(lowerCaseArg0);
+            final ActionArgument actionArgument = holder.getActionArgument();
+            parsedActionArgumentList.add(actionArgument);
+        }
+
+        int newArgsStartPos = 1;
+        if (2 <= args.length)
+        {
+            final ActionArgumentType actionArgumentType = holder.getActionArgument().getType();
+            // start by trying to parse the maximum amount of arguments allowed
+            int potentialArgumentLength = Math.min(args.length, actionArgumentType.allowedSpaceCount() + 1);
+            boolean requiresExactMatch = false;
+            
+            // build string of all arguments up to word count and check if they parse or provide tab completion
+            while (potentialArgumentLength > 0)
+            {
+                final StringBuilder sb = new StringBuilder(args[1]);
+                for (int i = 2; i < potentialArgumentLength; i++)
+                {
+                    final String nextPotentialWord = args[i];
+                    sb.append(' ').append(nextPotentialWord);
+                }
+
+                final String potentialArgumentValue = sb.toString();
+                final Object parsedObject = actionArgumentType.parse(server, sender, pos, moduleContext, actionMenuState, potentialArgumentValue);
+                if (null == parsedObject)
+                {
+                    final List<String> tabCompletions = actionArgumentType.getTabCompletions(server, pos, moduleContext, actionMenuState, potentialArgumentValue);
+                    if (!tabCompletions.isEmpty())
+                    {
+                        if (!requiresExactMatch || (1 == tabCompletions.size()))
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Merge this if statement with the enclosing one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1066)